src/lightning_fabric/fabric.py

```diff
@@ -518,6 +521,44 @@ def launch(self, function: Optional[Callable[["Fabric"], Any]] = None, *args: An
             return self._strategy.launcher.launch(function, *args, **kwargs)
         return function(*args, **kwargs)
 
+    def call(self, hook_name: str, *args: Any, **kwargs: Any) -> None:
+        """Trigger the callback methods with the given name and arguments.
+
+        Not all objects registered via ``Fabric(callbacks=...)`` must implement a method with the given name. The ones
+        that have a matching method name will get called.
+
+        Args:
+            hook_name: The name of the callback method.
+            *args: Optional positional arguments that get passed down to the callback method.
+            **kwargs: Optional keyword arguments that get passed down to the callback method.
+
+        Example::
+
+            class MyCallback:
+                def on_train_epoch_end(self, results):
+                    ...
+
+            fabric = Fabric(callbacks=[MyCallback]))
+            fabric.call("on_train_epoch_end", results={...})
+        """
+        for callback in self._callbacks:
+            method = getattr(callback, hook_name, None)
+            if method is None:
+                continue
+            if not callable(method):
+                rank_zero_warn(
+                    f"Skipping the callback `{type(callback).__name__}.{hook_name}` because it is not callable."
+                )
+                continue
+
+            method(*args, **kwargs)
```

Comment:
could we do something like `return_types = [callback.method(*args, **kwargs) for callback in self._callbacks]`? So that you don't need to store it in internal state if you don't want to?

Reply:
I think we should add this based on need, in a future iteration.

Reply:
@carmocca I agree that dict-like would be great, but then you'd also need a dict-like structure for your callbacks and in practice, execution-order of callbacks could be important although it shouldn't.

@awaelchli agreed!